include/mbgl/util/overloaded.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace mbgl
+{
+namespace util
+{
+    
+template <typename... Ts>
```

Comment:
Everything in `include` will become part of our public API. Is this intended? Can you add some triple slash comments to indicate what this template struct is for?

Reply:
agreed! I believe there's not need for it to be public but I have to check

Reply:
done, I moved it to src/mbgl/util